mosdef_cassandra/core/moves.py

```diff
@@ -173,6 +181,94 @@ def __init__(self, ensemble, species_topologies):
                 self.prob_rotate = 0.0
 
+    @property
+    def restricted_type(self):
+        return self._restricted_type
+
+    @restricted_type.setter
+    def restricted_type(self, restricted_type):
+        if restricted_type:
+            if self.ensemble == 'gcmc':
+                if len(restricted_type) != 1:
+                    raise ValueError("{} ensemble only has 1 box"
+                    ' but restricted_insertion of length {}'
+                    ' was passed.'.format(
+                        self.ensemble,
+                        len(restricted_type)))
+                if not isinstance(restricted_type[0], (list, tuple)):
+                    raise TypeError("'restricted_type' specified for"
+                    " a box must be an iterable.")
+                if len(restricted_type[0]) != self._n_species:
+                    raise ValueError("{} species specified"
+                    ' but restricted_type for {}'
+                    ' species was passed.'.format(
+                        self._n_species,
+                        len(restricted_type[0])))
+            elif self.ensemble in ['gemc', 'gemc_npt']:
+                if len(restricted_type) != 2:
+                    raise ValueError("{} ensemble requires 2 boxes"
+                    ' but restricted_type of length {}'
+                    ' was passed.'.format(
+                        self.ensemble,
+                        len(restricted_type)))
+                if len(restricted_type[0]) != self._n_species:
+                    raise ValueError("{} species specified"
+                    ' but restricted_typed for only {}'
+                    ' species was passed.'.format(
+                        self._n_species,
+                        len(restricted_type[0])))
+            else:
+                raise ValueError("Restricted insertions are only valid"
+                    ' for GCMC, GEMC, and GEMC-NPT ensembles.'
+                    ' The {} ensemble was passed.'.format(
+                        self.ensemble))
+
+            self._restricted_type = restricted_type
+        else:
+            self._restricted_type = restricted_type
+
+    @property
+    def restricted_value(self):
+        return self._restricted_value
+
+    @restricted_value.setter
+    def restricted_value(self, restricted_value):
+        if restricted_value:
+            if len(self.restricted_type) != len(restricted_value):
+                raise ValueError("'restricted_type' is a list of length"
+                        " {} but 'restricted_value' is a list of length"
+                        " {}.".format(
+                            len(self.restricted_type),
+                            len(restricted_value))
+                        )
+            if not isinstance(restricted_value[0], (list, tuple)):
+                raise TypeError("'restricted_value' specified for"
+                " a box must be an iterable.")
```

Comment:
The "must be an iterable" terminology, while technically correct, is, perhaps a bit confusing and not to-the-point. Something about a list or tuple with one element per species would be more clear IMO.